### ISSUE TITLE
Traininer achieves low error rate, and then error rate goes back up to a very high level

### ISSUE THREAD

**Issue by napowell** [Microsoft/CNTK#1000]
See below the output for part of the first epoch of training a LSTM-RNN sequence to sequence model with two hidden layers for the encoder and decoder each. The learning rate is set at 0.007 and the momentum at 1100. The trainer quickly gets an error rate below 10%, and then the error rate gets a lot worse. What is your recommendation for improving the training parameters, based on your experience? I tried halving the momentum, but then I didn't get the low error rate, and the error rate seemed to just dip down to about 85% and then go back up to 88% and get stuck there just like the run below with 1100 momentum. 

Find attached my config file.
[UniversalTriggeringShare.txt](https://github.com/Microsoft/CNTK/files/572655/UniversalTriggeringShare.txt)

11/04/2016 18:59:02:  Epoch[ 1 of 50]-Minibatch[   1-   1]: ce = 5.38877695 * 948; errs = 100.000% * 948; time = 4.9724s; samplesPerSecond = 190.7
11/04/2016 18:59:02:  Epoch[ 1 of 50]-Minibatch[   2-   2]: ce = 5.11416687 * 945; errs = 88.889% * 945; time = 0.6199s; samplesPerSecond = 1524.3
11/04/2016 18:59:03:  Epoch[ 1 of 50]-Minibatch[   3-   3]: ce = 7.49044187 * 946; errs = 88.901% * 946; time = 0.5714s; samplesPerSecond = 1655.5
11/04/2016 18:59:03:  Epoch[ 1 of 50]-Minibatch[   4-   4]: ce = 6.67617922 * 957; errs = 95.402% * 957; time = 0.5172s; samplesPerSecond = 1850.2
11/04/2016 18:59:04:  Epoch[ 1 of 50]-Minibatch[   5-   5]: ce = 5.71761308 * 950; errs = 97.474% * 950; time = 0.5228s; samplesPerSecond = 1817.2
11/04/2016 18:59:04:  Epoch[ 1 of 50]-Minibatch[   6-   6]: ce = 8.27089390 * 946; errs = 93.763% * 946; time = 0.5172s; samplesPerSecond = 1828.9
11/04/2016 18:59:05:  Epoch[ 1 of 50]-Minibatch[   7-   7]: ce = 4.75823602 * 950; errs = 91.789% * 950; time = 0.4818s; samplesPerSecond = 1971.9
11/04/2016 18:59:06:  Epoch[ 1 of 50]-Minibatch[   8-   8]: ce = 3.72795936 * 948; errs = 89.030% * 948; time = 0.6051s; samplesPerSecond = 1566.7
11/04/2016 18:59:06:  Epoch[ 1 of 50]-Minibatch[   9-   9]: ce = 3.42148022 * 940; errs = 82.340% * 940; time = 0.4994s; samplesPerSecond = 1882.3
WARNING: The same matrix with dim [1, 1050] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
WARNING: The same matrix with dim [1, 21] has been transferred between different devices for 20 times.
11/04/2016 18:59:07:  Epoch[ 1 of 50]-Minibatch[  10-  10]: ce = 2.99753485 * 946; errs = 79.915% * 946; time = 0.5010s; samplesPerSecond = 1888.1
11/04/2016 18:59:57:  Epoch[ 1 of 50]-Minibatch[  11- 100]: ce = 0.60057173 * 83592; errs = 14.674% * 83592; time = 50.4277s; samplesPerSecond = 1657.7
11/04/2016 19:00:52:  Epoch[ 1 of 50]-Minibatch[ 101- 200]: ce = 0.34388031 * 93345; errs = 9.759% * 93345; time = 55.4981s; samplesPerSecond = 1681.9
11/04/2016 19:01:48:  Epoch[ 1 of 50]-Minibatch[ 201- 300]: ce = 0.30624595 * 92625; errs = 8.937% * 92625; time = 55.3791s; samplesPerSecond = 1672.6
11/04/2016 19:02:42:  Epoch[ 1 of 50]-Minibatch[ 301- 400]: ce = 0.29943440 * 93143; errs = 8.677% * 93143; time = 54.4049s; samplesPerSecond = 1712.0
11/04/2016 19:03:41:  Epoch[ 1 of 50]-Minibatch[ 401- 500]: ce = 0.29382401 * 91502; errs = 8.706% * 91502; time = 58.9981s; samplesPerSecond = 1550.9
11/04/2016 19:04:35:  Epoch[ 1 of 50]-Minibatch[ 501- 600]: ce = 0.28736255 * 93670; errs = 8.530% * 93670; time = 53.4996s; samplesPerSecond = 1750.9
11/04/2016 19:05:32:  Epoch[ 1 of 50]-Minibatch[ 601- 700]: ce = 0.28511490 * 92855; errs = 8.436% * 92855; time = 56.8158s; samplesPerSecond = 1634.3
11/04/2016 19:06:28:  Epoch[ 1 of 50]-Minibatch[ 701- 800]: ce = 0.28215272 * 92152; errs = 8.371% * 92152; time = 56.2487s; samplesPerSecond = 1638.3
11/04/2016 19:07:22:  Epoch[ 1 of 50]-Minibatch[ 801- 900]: ce = 0.27472080 * 93838; errs = 8.162% * 93838; time = 54.5177s; samplesPerSecond = 1721.2
11/04/2016 19:08:16:  Epoch[ 1 of 50]-Minibatch[ 901-1000]: ce = 0.27792223 * 94010; errs = 8.285% * 94010; time = 53.2090s; samplesPerSecond = 1766.8
11/04/2016 19:09:08:  Epoch[ 1 of 50]-Minibatch[1001-1100]: ce = 0.27787087 * 93848; errs = 8.378% * 93848; time = 52.8819s; samplesPerSecond = 1774.7
11/04/2016 19:10:02:  Epoch[ 1 of 50]-Minibatch[1101-1200]: ce = 0.27814439 * 93792; errs = 8.285% * 93792; time = 53.3611s; samplesPerSecond = 1757.7
11/04/2016 19:10:56:  Epoch[ 1 of 50]-Minibatch[1201-1300]: ce = 0.27775835 * 93473; errs = 8.385% * 93473; time = 54.3632s; samplesPerSecond = 1719.4
11/04/2016 19:11:54:  Epoch[ 1 of 50]-Minibatch[1301-1400]: ce = 2.08702128 * 92133; errs = 53.984% * 92133; time = 58.1290s; samplesPerSecond = 1585.0
11/04/2016 19:12:51:  Epoch[ 1 of 50]-Minibatch[1401-1500]: ce = 3.41095847 * 92692; errs = 88.872% * 92692; time = 57.0538s; samplesPerSecond = 1624.6
11/04/2016 19:13:47:  Epoch[ 1 of 50]-Minibatch[1501-1600]: ce = 3.41283145 * 92737; errs = 88.877% * 92737; time = 56.1030s; samplesPerSecond = 1653.0
11/04/2016 19:14:40:  Epoch[ 1 of 50]-Minibatch[1601-1700]: ce = 3.41043888 * 94081; errs = 88.877% * 94081; time = 52.8866s; samplesPerSecond = 1778.9
11/04/2016 19:15:35:  Epoch[ 1 of 50]-Minibatch[1701-1800]: ce = 3.41004508 * 93048; errs = 88.871% * 93048; time = 54.4373s; samplesPerSecond = 1709.3
11/04/2016 19:16:30:  Epoch[ 1 of 50]-Minibatch[1801-1900]: ce = 3.40862513 * 93100; errs = 88.878% * 93100; time = 55.1569s; samplesPerSecond = 1687.9
11/04/2016 19:17:23:  Epoch[ 1 of 50]-Minibatch[1901-2000]: ce = 3.40987158 * 94063; errs = 88.874% * 94063; time = 53.5776s; samplesPerSecond = 1755.6
11/04/2016 19:18:19:  Epoch[ 1 of 50]-Minibatch[2001-2100]: ce = 3.40681808 * 92966; errs = 88.872% * 92966; time = 55.2804s; samplesPerSecond = 1681.7
11/04/2016 19:19:13:  Epoch[ 1 of 50]-Minibatch[2101-2200]: ce = 3.40483567 * 93772; errs = 88.867% * 93772; time = 54.5270s; samplesPerSecond = 1719.7
11/04/2016 19:20:09:  Epoch[ 1 of 50]-Minibatch[2201-2300]: ce = 3.40779951 * 92025; errs = 88.862% * 92025; time = 55.7001s; samplesPerSecond = 1652.2
11/04/2016 19:21:03:  Epoch[ 1 of 50]-Minibatch[2301-2400]: ce = 3.41005042 * 94011; errs = 88.890% * 94011; time = 54.1468s; samplesPerSecond = 1736.2
11/04/2016 19:21:59:  Epoch[ 1 of 50]-Minibatch[2401-2500]: ce = 3.41051798 * 92784; errs = 88.877% * 92784; time = 55.6725s; samplesPerSecond = 1666.6
11/04/2016 19:22:55:  Epoch[ 1 of 50]-Minibatch[2501-2600]: ce = 3.40847928 * 92897; errs = 88.880% * 92897; time = 56.6255s; samplesPerSecond = 1640.6
11/04/2016 19:23:50:  Epoch[ 1 of 50]-Minibatch[2601-2700]: ce = 3.40919616 * 93713; errs = 88.881% * 93713; time = 54.3897s; samplesPerSecond = 1723.0
11/04/2016 19:24:44:  Epoch[ 1 of 50]-Minibatch[2701-2800]: ce = 3.40861402 * 93313; errs = 88.876% * 93313; time = 53.7431s; samplesPerSecond = 1736.3
11/04/2016 19:25:38:  Epoch[ 1 of 50]-Minibatch[2801-2900]: ce = 3.40853014 * 93703; errs = 88.890% * 93703; time = 54.8749s; samplesPerSecond = 1707.6

**Comment by n17s** [comment]
Set the epochSize to 1300000 (last good minibatch * minibatchSize) and specify your learning rate as 0.007:0.003 (halve the learning rate after the first epoch). If half still diverges immediately, try smaller gradient clipping. If it diverges after a while try to halve the learning rate again on the last epoch before you see divergence (you might have to redefine epochSize to a smaller value). Also you might only need to adjust the epochSize and let the AutoAdjust block find a good learning rate.
